src/connectors/crowdstrike_devices.py

```diff
@@ -256,8 +308,258 @@ def ingest(table_name, options):
                 )
                 for device in devices
             ],
-            select=db.derive_insert_select(LANDING_TABLE_COLUMNS),
-            columns=db.derive_insert_columns(LANDING_TABLE_COLUMNS),
+            select=db.derive_insert_select(LANDING_DEVICES_TABLE_COLUMNS),
+            columns=db.derive_insert_columns(LANDING_DEVICES_TABLE_COLUMNS),
         )
         log.info(f'Inserted {len(devices)} rows.')
         yield len(devices)
+
+
+rem_id_cache = {}
+
+
+async def does_rem_exist(rem_id: str) -> str:
+    """
+    does_rem_exist returns "" if the remediation is has already been cached locally
+    or is recorded in the database. Otherwise, it returns the remediation ID.
+    """
+    # Check local cache if you have seen this ID in this run
+    if rem_id in rem_id_cache:
+        log.info(f"[CACHE]: Local cache hit! Already recorded {rem_id} in this round")
+        return ""
+
+    # Check if we have it stored in the table already
+    row = db.fetch_latest(
+        "data.CROWDSTRIKE_SPOTLIGHT_REMS_DEFAULT_CONNECTION",
```

Comment:
This probably needs to be parameterized like [these](https://github.com/snowflakedb/SnowAlert/pull/489/files#diff-0f3bcae7b198de7ad264a1810da509be9215dbd8427ab4501feb5352144219fdR185)

Reply:
So are we not worrying about SQL injection here? Looks like the DB library has a prepared statements method but I don't see many people using it.

Reply:
Also, do you know how the connection name stuff works? I only saw the string "DEFAULT", but are there other strings?